### DISCUSSION TITLE
DataSocket to Python

### DISCUSSION THREAD

**Discussion by FabiM17** [community/community#131145] [New to GitHub]
Hi, 

im trying to connect to a NI DataSocket which is implemented into an exe file. Ich want to transfer Data from the DataSocket Server to my Python Skript. 

There is an additional DataSocket Receiver exe which is working perfect.  But my Python Skript wont. If i open up the NI DataSocket Server i can see that my Python Skript connects as additional process an but i receive 0 bytes.

Python:
Connected to DataSocket Server at dstp://localhost/WFS_WavefrontSensor
Data value type: <class 'int'>
Data value content: 0
Data value is not in the expected dictionary format or is empty

I just dont know how the DataSocket communicates with the receiver. Is there any help?

Please let me know your suggestions

Thank you in advance 
Fabian

**Comment by chiarorosa** [comment]
@FabiM17 try this for testing, is a simple example to illustrate connecting and reading data from a DataSocket server in Python using `pyvisa`:

```python
import pyvisa

# Establish connection to the DataSocket server
rm = pyvisa.ResourceManager()
ds = rm.open_resource('DSTP::localhost::WFS_WavefrontSensor')

# Read data from the server
try:
    data = ds.read()
    print(f"Data received: {data}")
except Exception as e:
    print(f"Error reading data: {e}")
finally:
    ds.close()
```

Ensure you have the necessary NI drivers and libraries installed on your system. If `pyvisa` is not suitable for your case, you might need to use other libraries or interfaces provided by NI.

**Comment by FabiM17** [comment]
My Problem: Unable to Receive Data from existing DataSocket Server.

The server is running on a exe file, where i can choose the url path: e.g. dstp://localhost/WFS_WavefrontSensor

The Server is working, with the client exe of the manufacturer. I just want to implement the same for my python script and read the data from the server and still run the exe for adjustments. Is there any possibility with a python lib.

While the connection is successfully established and requests are sent, I am not receiving any payload data back from the server. Here are the debug messages I receive: `` Connected to DataSocket server. Sent update request to server. Received raw data: b'' No data received on update.
It’s possible that the requests may not match the server's expected format, or the server is not responding as anticipated. I have attached a wireshark screenshot

If you have any suggestions please let me know. Thank you!

![Screenshot_wireshark](https://github.com/community/community/assets/174805725/a2a4a560-73a5-45cb-ae83-346f22eda6d1)
